tests/pipeline/test_pipeline.py

```diff
@@ -323,6 +323,76 @@ def test_combine(self):
+                node(biconcat, ["input", "input2"], "output2", name="b"),
+                node(biconcat, ["input", "input3"], "output3", name="c"),
+            ]
+        )
+        pipeline2 = Pipeline([node(biconcat, ["input", "input2"], "output2", name="b")])
+        new_pipeline = pipeline1 - pipeline2
+        assert new_pipeline.inputs() == {"input", "input1", "input3"}
+        assert new_pipeline.outputs() == {"output1", "output3"}
+        assert {n.name for n in new_pipeline.nodes} == {"a", "c"}
+
+    def test_remove_with_partial_intersection(self):
+        """Create a pipeline of 3 nodes and remove one of them using a pipeline
+        that contains a partial match.
+        """
+        pipeline1 = Pipeline(
+            [
+                node(biconcat, ["input", "input1"], "output1", name="a"),
+                node(biconcat, ["input", "input2"], "output2", name="b"),
+                node(biconcat, ["input", "input3"], "output3", name="c"),
+            ]
+        )
+        pipeline2 = Pipeline(
+            [
+                node(biconcat, ["input", "input2"], "output2", name="b"),
+                node(biconcat, ["input", "input4"], "output4", name="d"),
+            ]
+        )
+        new_pipeline = pipeline1 - pipeline2
+        assert new_pipeline.inputs() == {"input", "input1", "input3"}
+        assert new_pipeline.outputs() == {"output1", "output3"}
+        assert {n.name for n in new_pipeline.nodes} == {"a", "c"}
+
+    def test_remove_empty_from_pipeline(self):
+        """Remove an empty pipeline"""
+        pipeline1 = Pipeline([node(biconcat, ["input", "input1"], "output1", name="a")])
+        pipeline2 = Pipeline([])
+        new_pipeline = pipeline1 - pipeline2
+        assert new_pipeline.inputs() == pipeline1.inputs()
+        assert new_pipeline.outputs() == pipeline1.outputs()
+        assert {n.name for n in new_pipeline.nodes} == {"a"}
+
+    def test_remove_from_empty_pipeline(self):
+        """Remove node from an empty pipeline"""
+        pipeline1 = Pipeline([node(biconcat, ["input", "input1"], "output1", name="a")])
+        pipeline2 = Pipeline([])
+        new_pipeline = pipeline2 - pipeline1
+        assert new_pipeline.inputs() == pipeline2.inputs()
+        assert new_pipeline.outputs() == pipeline2.outputs()
+        assert not new_pipeline.nodes
+
+    def test_remove_all_nodes(self):
+        """Remove an entire pipeline"""
+        pipeline1 = Pipeline([node(biconcat, ["input", "input1"], "output1", name="a")])
+        pipeline2 = Pipeline([node(biconcat, ["input", "input1"], "output1", name="a")])
+        new_pipeline = pipeline1 - pipeline2
+        assert new_pipeline.inputs() == set()
+        assert new_pipeline.outputs() == set()
+        assert not new_pipeline.nodes
+
+    def test_invalid_remove(self):
+        p = Pipeline([])
+        pattern = r"unsupported operand type\(s\) for -: 'Pipeline' and 'str'"
+        with pytest.raises(TypeError, match=pattern):
```

Comment:
Isn't that supposed to raise `NotImplemented` exception?

Reply:
It does seem a bit odd at first, I stole the pattern from `invalid_intersection`, `invalid_union`, and `invalid_add`.

Reply:
Found this as a good resource: https://stackoverflow.com/questions/878943/why-return-notimplemented-instead-of-raising-notimplementederror